File: frontend/src/types/savedWidget.ts
export interface SavedWidgetRequest {
    title: string;
    countries: string;
    indicatorCode: string;
    chartType: string;
    startYear: number;
    endYear: number;
}

export interface SavedWidgetResponse {
    id: number;
    title: string;
    countries: string;
    indicatorCode: string;
    chartType: string;
    startYear: number;
    endYear: number;
}

File: backend/src/main/java/com/worldmetrics/backend/model/SavedWidget.java
package com.worldmetrics.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "saved_widgets")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SavedWidget {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String countries;

    @Column(name = "indicator_code", nullable = false, length = 50)
    private String indicatorCode;

    @Column(name = "chart_type", nullable = false, length = 50)
    private String chartType;

    @Column(name = "start_year")
    private Integer startYear;

    @Column(name = "end_year")
    private Integer endYear;
}